{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA SEQ ANALYSIS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parts of this code come from the sambomics tutorial: https://sanbomics.com/2022/01/08/complete-rnaseq-alignment-guide-from-fastq-to-count-table/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1ยบ FASTQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "            FastQC - A high throughput sequence QC analysis tool\n",
      "\n",
      "SYNOPSIS\n",
      "\n",
      "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
      "\n",
      "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
      "           [-c contaminant file] seqfile1 .. seqfileN\n",
      "\n",
      "DESCRIPTION\n",
      "\n",
      "    FastQC reads a set of sequence files and produces from each one a quality\n",
      "    control report consisting of a number of different modules, each one of \n",
      "    which will help to identify a different potential type of problem in your\n",
      "    data.\n",
      "    \n",
      "    If no files to process are specified on the command line then the program\n",
      "    will start as an interactive graphical application.  If files are provided\n",
      "    on the command line then the program will run with no user interaction\n",
      "    required.  In this mode it is suitable for inclusion into a standardised\n",
      "    analysis pipeline.\n",
      "    \n",
      "    The options for the program as as follows:\n",
      "    \n",
      "    -h --help       Print this help file and exit\n",
      "    \n",
      "    -v --version    Print the version of the program and exit\n",
      "    \n",
      "    -o --outdir     Create all output files in the specified output directory.\n",
      "                    Please note that this directory must exist as the program\n",
      "                    will not create it.  If this option is not set then the \n",
      "                    output file for each sequence file is created in the same\n",
      "                    directory as the sequence file which was processed.\n",
      "                    \n",
      "    --casava        Files come from raw casava output. Files in the same sample\n",
      "                    group (differing only by the group number) will be analysed\n",
      "                    as a set rather than individually. Sequences with the filter\n",
      "                    flag set in the header will be excluded from the analysis.\n",
      "                    Files must have the same names given to them by casava\n",
      "                    (including being gzipped and ending with .gz) otherwise they\n",
      "                    won't be grouped together correctly.\n",
      "                    \n",
      "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
      "                    this mode you can pass in directories to process and the program\n",
      "                    will take in all fast5 files within those directories and produce\n",
      "                    a single output file from the sequences found in all files.                    \n",
      "                    \n",
      "    --nofilter      If running with --casava then don't remove read flagged by\n",
      "                    casava as poor quality when performing the QC analysis.\n",
      "                   \n",
      "    --extract       If set then the zipped output file will be uncompressed in\n",
      "                    the same directory after it has been created. If --delete is \n",
      "                    also specified then the zip file will be removed after the \n",
      "                    contents are unzipped. \n",
      "                    \n",
      "    -j --java       Provides the full path to the java binary you want to use to\n",
      "                    launch fastqc. If not supplied then java is assumed to be in\n",
      "                    your path.\n",
      "                   \n",
      "    --noextract     Do not uncompress the output file after creating it.  You\n",
      "                    should set this option if you do not wish to uncompress\n",
      "                    the output when running in non-interactive mode.\n",
      "                    \n",
      "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
      "                    show data for every base in the read.  WARNING: Using this\n",
      "                    option will cause fastqc to crash and burn if you use it on\n",
      "                    really long reads, and your plots may end up a ridiculous size.\n",
      "                    You have been warned!\n",
      "                    \n",
      "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
      "                    to be shown in the report.  As long as you set this to a value\n",
      "                    greater or equal to your longest read length then this will be\n",
      "                    the sequence length used to create your read groups.  This can\n",
      "                    be useful for making directly comaparable statistics from \n",
      "                    datasets with somewhat variable read lengths.\n",
      "\n",
      "    --dup_length    Sets a length to which the sequences will be truncated when \n",
      "                    defining them to be duplicates, affecting the duplication and\n",
      "                    overrepresented sequences plot.  This can be useful if you have\n",
      "                    long reads with higher levels of miscalls, or contamination with\n",
      "                    adapter dimers containing UMI sequences.\n",
      "\n",
      "                    \n",
      "    -f --format     Bypasses the normal sequence file format detection and\n",
      "                    forces the program to use the specified format.  Valid\n",
      "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
      "                    \n",
      "\n",
      "    --memory        Sets the base amount of memory, in Megabytes, used to process \n",
      "                    each file.  Defaults to 512MB.  You may need to increase this if\n",
      "                    you have a file with very long sequences in it.\n",
      "                \n",
      "    --svg           Save the graphs in the report in SVG format.\n",
      "\n",
      "    -t --threads    Specifies the number of files which can be processed\n",
      "                    simultaneously.  Each thread will be allocated 250MB of\n",
      "                    memory so you shouldn't run more threads than your\n",
      "                    available memory will cope with, and not more than\n",
      "                    6 threads on a 32 bit machine\n",
      "                  \n",
      "    -c              Specifies a non-default file which contains the list of\n",
      "    --contaminants  contaminants to screen overrepresented sequences against.\n",
      "                    The file must contain sets of named contaminants in the\n",
      "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
      "                    be ignored.\n",
      "\n",
      "    -a              Specifies a non-default file which contains the list of\n",
      "    --adapters      adapter sequences which will be explicity searched against\n",
      "                    the library. The file must contain sets of named adapters\n",
      "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
      "                    will be ignored.\n",
      "                    \n",
      "    -l              Specifies a non-default file which contains a set of criteria\n",
      "    --limits        which will be used to determine the warn/error limits for the\n",
      "                    various modules.  This file can also be used to selectively \n",
      "                    remove some modules from the output all together.  The format\n",
      "                    needs to mirror the default limits.txt file found in the\n",
      "                    Configuration folder.\n",
      "                    \n",
      "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
      "                    module. Specified Kmer length must be between 2 and 10. Default\n",
      "                    length is 7 if not specified.\n",
      "                    \n",
      "   -q --quiet       Suppress all progress messages on stdout and only report errors.\n",
      "   \n",
      "   -d --dir         Selects a directory to be used for temporary files written when\n",
      "                    generating report images. Defaults to system temp directory if\n",
      "                    not specified.\n",
      "                    \n",
      "BUGS\n",
      "\n",
      "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
      "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
      "                   \n",
      "    \n"
     ]
    }
   ],
   "source": [
    "!fastqc -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The command used was:\n",
    "\n",
    "fastqc --extract\\\n",
    "--delete\\\n",
    "--nogroup\\\n",
    "-o /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/fastqc\\\n",
    "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports//all_files/*.fq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "When the fastqc had run we could take a look to the output:\n",
    "\n",
    "This are two plots from the report of a single file (as an example) but all of them looked like that:\n",
    "\n",
    "- Here we can see that the quality is good along all the sequences and sequence position.\n",
    "\n",
    "\n",
    "![](./images/PB_sequence_quality.png)\n",
    "\n",
    "\n",
    "- Here we can see that the per base sequence content is normal along all the sequence until the 13th first bases, this deviation in the expected base frecquency suggest the pressence of adapters.\n",
    "\n",
    "\n",
    "![](./images/PB_sequence_content.png)\n",
    "\n",
    "\n",
    "\n",
    "- Thus, the next step will be a trimming with trim-galore. We will perform a hard timming to remove the adapters in the 13 fist positions. We wont conduct a quality or adapter trimming because we new that the quality is great. Also, we dont need to infer the adapters or to find them from a predefined adapter list, we can just remove the starting part of the sequences.\n",
    "\n",
    "The command used was: \n",
    "\n",
    "trim_galore -o /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files\\ # Output dir\n",
    "            -j 8 \\ # Number of cores recommended \n",
    "            --hardtrim3 137 \\ # Length of the secuences after trimming (trim the first 13 bases)\n",
    "            /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/all_files/*.fq.gz  # path to the fq files\n",
    "\n",
    "\n",
    "First I had to create a new enviroment to run trim-galore due to dependencie issues:\n",
    "\n",
    "conda create -n trimming trim-galore\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**INDEX BUILDING:**\n",
    "\n",
    "Then I had to build an index for the reference genome that STAR can use for the alignment.\n",
    "\n",
    "- First I downloaded the FASTA and GTF files of the Human genome (GRCh38) from ensembl: https://www.ensembl.org/info/data/ftp/index.html\n",
    "\n",
    "- Then I run the following command:\n",
    "\n",
    "STAR --runMode genomeGenerate \\\n",
    "--genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ \\\n",
    "--genomeFastaFiles /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa \\\n",
    "--sjdbGTFfile /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/Homo_sapiens.GRCh38.110.gtf \\\n",
    "--runThreadN 30"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Actually perform the alignment:**\n",
    "\n",
    "For the alignement I will take a manifest table (structure: [fq1 \\t fq2 \\t ID]) that contains the path info of the paired reads and a name for the sample, then I will iterate the rows of these table and run the STAR alignment command for the paired files of each row. \n",
    "\n",
    "- The command is:\n",
    "\n",
    "STAR --runMode alignReads \\\n",
    "--runThreadN 30\\\n",
    "--genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ \\\n",
    "--outSAMtype BAM SortedByCoordinate \\\n",
    "--outFileNamePrefix {prefix}\\\n",
    "--readFilesIn {fq1} {fq2}\n",
    "--readFilesCommand zcat\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/11_1.137bp_3prime.fq.gz',\n",
       " '/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/11_2.137bp_3prime.fq.gz',\n",
       " 'ID: sample 11')"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Charge the manifest file used to indicate where are the paired reads \n",
    "import pandas as pd\n",
    "\n",
    "manifest = pd.read_csv(\"./clean_data_reports/manifest.tsv\",sep=\"\\t\",header=None)\n",
    "manifest[0][0],manifest[1][0],manifest[2][0] # Show the row info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/11_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/11_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_11 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_11 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/11_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/11_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 16:35:42 ..... started STAR run\n",
      "Dec 07 16:35:42 ..... loading genome\n",
      "Dec 07 16:36:10 ..... started mapping\n",
      "Dec 07 16:39:02 ..... finished mapping\n",
      "Dec 07 16:39:04 ..... started sorting BAM\n",
      "Dec 07 16:40:12 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/15_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/15_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_15 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_15 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/15_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/15_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 16:40:16 ..... started STAR run\n",
      "Dec 07 16:40:16 ..... loading genome\n",
      "Dec 07 16:40:25 ..... started mapping\n",
      "Dec 07 16:43:55 ..... finished mapping\n",
      "Dec 07 16:43:56 ..... started sorting BAM\n",
      "Dec 07 16:45:05 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/20_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/20_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_20 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_20 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/20_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/20_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 16:45:07 ..... started STAR run\n",
      "Dec 07 16:45:07 ..... loading genome\n",
      "Dec 07 16:45:15 ..... started mapping\n",
      "Dec 07 16:48:38 ..... finished mapping\n",
      "Dec 07 16:48:39 ..... started sorting BAM\n",
      "Dec 07 16:49:48 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/12_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/12_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_12 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_12 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/12_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/12_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 16:49:51 ..... started STAR run\n",
      "Dec 07 16:49:51 ..... loading genome\n",
      "Dec 07 16:49:59 ..... started mapping\n",
      "Dec 07 16:53:20 ..... finished mapping\n",
      "Dec 07 16:53:20 ..... started sorting BAM\n",
      "Dec 07 16:54:31 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/7_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/7_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_7 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_7 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/7_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/7_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 16:54:34 ..... started STAR run\n",
      "Dec 07 16:54:34 ..... loading genome\n",
      "Dec 07 16:54:41 ..... started mapping\n",
      "Dec 07 16:56:45 ..... finished mapping\n",
      "Dec 07 16:56:46 ..... started sorting BAM\n",
      "Dec 07 16:58:07 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/10_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/10_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_10 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_10 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/10_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/10_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 16:58:09 ..... started STAR run\n",
      "Dec 07 16:58:09 ..... loading genome\n",
      "Dec 07 16:58:17 ..... started mapping\n",
      "Dec 07 17:01:24 ..... finished mapping\n",
      "Dec 07 17:01:25 ..... started sorting BAM\n",
      "Dec 07 17:02:33 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/3_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/3_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_3 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_3 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/3_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/3_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:02:36 ..... started STAR run\n",
      "Dec 07 17:02:36 ..... loading genome\n",
      "Dec 07 17:02:43 ..... started mapping\n",
      "Dec 07 17:05:43 ..... finished mapping\n",
      "Dec 07 17:05:44 ..... started sorting BAM\n",
      "Dec 07 17:06:55 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/2_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/2_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_2 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_2 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/2_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/2_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:06:57 ..... started STAR run\n",
      "Dec 07 17:06:57 ..... loading genome\n",
      "Dec 07 17:07:06 ..... started mapping\n",
      "Dec 07 17:09:59 ..... finished mapping\n",
      "Dec 07 17:10:00 ..... started sorting BAM\n",
      "Dec 07 17:11:10 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/17_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/17_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_17 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_17 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/17_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/17_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:11:12 ..... started STAR run\n",
      "Dec 07 17:11:12 ..... loading genome\n",
      "Dec 07 17:11:21 ..... started mapping\n",
      "Dec 07 17:14:28 ..... finished mapping\n",
      "Dec 07 17:14:29 ..... started sorting BAM\n",
      "Dec 07 17:15:47 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/9_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/9_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_9 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_9 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/9_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/9_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:15:49 ..... started STAR run\n",
      "Dec 07 17:15:49 ..... loading genome\n",
      "Dec 07 17:15:57 ..... started mapping\n",
      "Dec 07 17:17:58 ..... finished mapping\n",
      "Dec 07 17:17:59 ..... started sorting BAM\n",
      "Dec 07 17:19:14 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/19_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/19_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_19 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_19 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/19_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/19_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:19:17 ..... started STAR run\n",
      "Dec 07 17:19:17 ..... loading genome\n",
      "Dec 07 17:19:24 ..... started mapping\n",
      "Dec 07 17:22:44 ..... finished mapping\n",
      "Dec 07 17:22:45 ..... started sorting BAM\n",
      "Dec 07 17:23:51 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/18_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/18_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_18 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_18 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/18_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/18_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:23:54 ..... started STAR run\n",
      "Dec 07 17:23:54 ..... loading genome\n",
      "Dec 07 17:24:01 ..... started mapping\n",
      "Dec 07 17:27:10 ..... finished mapping\n",
      "Dec 07 17:27:11 ..... started sorting BAM\n",
      "Dec 07 17:28:24 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/24_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/24_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_24 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_24 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/24_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/24_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:28:27 ..... started STAR run\n",
      "Dec 07 17:28:27 ..... loading genome\n",
      "Dec 07 17:28:34 ..... started mapping\n",
      "Dec 07 17:31:34 ..... finished mapping\n",
      "Dec 07 17:31:35 ..... started sorting BAM\n",
      "Dec 07 17:32:50 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/6_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/6_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_6 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_6 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/6_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/6_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:32:53 ..... started STAR run\n",
      "Dec 07 17:32:53 ..... loading genome\n",
      "Dec 07 17:33:01 ..... started mapping\n",
      "Dec 07 17:36:00 ..... finished mapping\n",
      "Dec 07 17:36:01 ..... started sorting BAM\n",
      "Dec 07 17:37:11 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/22_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/22_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_22 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_22 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/22_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/22_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:37:13 ..... started STAR run\n",
      "Dec 07 17:37:13 ..... loading genome\n",
      "Dec 07 17:37:21 ..... started mapping\n",
      "Dec 07 17:40:18 ..... finished mapping\n",
      "Dec 07 17:40:19 ..... started sorting BAM\n",
      "Dec 07 17:41:29 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/8_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/8_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_8 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_8 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/8_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/8_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:41:31 ..... started STAR run\n",
      "Dec 07 17:41:31 ..... loading genome\n",
      "Dec 07 17:41:39 ..... started mapping\n",
      "Dec 07 17:43:41 ..... finished mapping\n",
      "Dec 07 17:43:42 ..... started sorting BAM\n",
      "Dec 07 17:44:56 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/5_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/5_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_5 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_5 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/5_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/5_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:44:59 ..... started STAR run\n",
      "Dec 07 17:44:59 ..... loading genome\n",
      "Dec 07 17:45:05 ..... started mapping\n",
      "Dec 07 17:47:59 ..... finished mapping\n",
      "Dec 07 17:48:00 ..... started sorting BAM\n",
      "Dec 07 17:49:14 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/14_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/14_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_14 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_14 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/14_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/14_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:49:16 ..... started STAR run\n",
      "Dec 07 17:49:16 ..... loading genome\n",
      "Dec 07 17:49:24 ..... started mapping\n",
      "Dec 07 17:52:38 ..... finished mapping\n",
      "Dec 07 17:52:39 ..... started sorting BAM\n",
      "Dec 07 17:53:45 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/23_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/23_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_23 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_23 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/23_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/23_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:53:48 ..... started STAR run\n",
      "Dec 07 17:53:48 ..... loading genome\n",
      "Dec 07 17:53:56 ..... started mapping\n",
      "Dec 07 17:56:58 ..... finished mapping\n",
      "Dec 07 17:56:59 ..... started sorting BAM\n",
      "Dec 07 17:58:08 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/21_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/21_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_21 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_21 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/21_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/21_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 17:58:11 ..... started STAR run\n",
      "Dec 07 17:58:11 ..... loading genome\n",
      "Dec 07 17:58:18 ..... started mapping\n",
      "Dec 07 18:01:27 ..... finished mapping\n",
      "Dec 07 18:01:28 ..... started sorting BAM\n",
      "Dec 07 18:02:43 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310003370_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310003370_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_1_8522310003370 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_1_8522310003370 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310003370_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310003370_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 18:02:46 ..... started STAR run\n",
      "Dec 07 18:02:46 ..... loading genome\n",
      "Dec 07 18:02:53 ..... started mapping\n",
      "Dec 07 18:06:09 ..... finished mapping\n",
      "Dec 07 18:06:10 ..... started sorting BAM\n",
      "Dec 07 18:07:15 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/4_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/4_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_4 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_4 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/4_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/4_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 18:07:18 ..... started STAR run\n",
      "Dec 07 18:07:18 ..... loading genome\n",
      "Dec 07 18:07:26 ..... started mapping\n",
      "Dec 07 18:10:27 ..... finished mapping\n",
      "Dec 07 18:10:28 ..... started sorting BAM\n",
      "Dec 07 18:11:44 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/13_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/13_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_13 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_13 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/13_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/13_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 18:11:47 ..... started STAR run\n",
      "Dec 07 18:11:47 ..... loading genome\n",
      "Dec 07 18:11:54 ..... started mapping\n",
      "Dec 07 18:14:54 ..... finished mapping\n",
      "Dec 07 18:14:55 ..... started sorting BAM\n",
      "Dec 07 18:16:10 ..... finished successfully\n",
      "/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310005559_1.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310005559_2.137bp_3prime.fq.gz \n",
      " /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_1_8522310005559 \n",
      " \n",
      " \n",
      "\n",
      "\t/home/victor/miniconda3/envs/rnaseq/bin/STAR-avx2 --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/sample_1_8522310005559 --readFilesIn /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310005559_1.137bp_3prime.fq.gz /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files/1_8522310005559_2.137bp_3prime.fq.gz --readFilesCommand zcat\n",
      "\tSTAR version: 2.7.11a   compiled: 2023-09-15T02:58:53+0000 :/opt/conda/conda-bld/star_1694746407721/work/source\n",
      "Dec 07 18:16:13 ..... started STAR run\n",
      "Dec 07 18:16:13 ..... loading genome\n",
      "Dec 07 18:16:20 ..... started mapping\n",
      "Dec 07 18:19:35 ..... finished mapping\n",
      "Dec 07 18:19:36 ..... started sorting BAM\n",
      "Dec 07 18:20:44 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(manifest)):\n",
    "    \n",
    "    fq1 = manifest.iloc[i,0]\n",
    "    fq2 = manifest.iloc[i,1]\n",
    "    pref = manifest.iloc[i,2].replace(\"ID: \",\"\").replace(\" \",\"_\")\n",
    "    prefix= f\"/media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/{pref}\"\n",
    "    print(fq1,\"\\n\",fq2,\"\\n\",prefix,\"\\n\",\"\\n\",\"\\n\",)\n",
    "    !STAR --runMode alignReads --runThreadN 30 --genomeDir /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/genome_index/ --outSAMtype BAM SortedByCoordinate --outFileNamePrefix {prefix} --readFilesIn {fq1} {fq2} --readFilesCommand zcat\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**FEATURE COUNTS:**\n",
    "\n",
    "Finally we will run featurecounts that will output a counts table which will be the starting point for the downstream analysis (differential expression, functional analysis...)\n",
    "\n",
    "- Installation: \n",
    "\n",
    "conda install -c bioconda subread\n",
    "\n",
    "- Command:\n",
    "\n",
    "featureCounts -a /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/GENOME_FILES/Homo_sapiens.GRCh38.110.gtf \\\n",
    "-o /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/featurecounts/counts.tsv \\\n",
    "-T 30 -p -B -C /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment/*.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "featureCounts: invalid option -- 'h'\n",
      "\n",
      "Version 2.0.6\n",
      "\n",
      "Usage: featureCounts [options] -a <annotation_file> -o <output_file> input_file1 [input_file2] ... \n",
      "\n",
      "## Mandatory arguments:\n",
      "\n",
      "  -a <string>         Name of an annotation file. GTF/GFF format by default. See\n",
      "                      -F option for more format information. Inbuilt annotations\n",
      "                      (SAF format) is available in 'annotation' directory of the\n",
      "                      package. Gzipped file is also accepted.\n",
      "\n",
      "  -o <string>         Name of output file including read counts. A separate file\n",
      "                      including summary statistics of counting results is also\n",
      "                      included in the output ('<string>.summary'). Both files\n",
      "                      are in tab delimited format.\n",
      "\n",
      "  input_file1 [input_file2] ...   A list of SAM or BAM format files. They can be\n",
      "                      either name or location sorted. If no files provided,\n",
      "                      <stdin> input is expected. Location-sorted paired-end reads\n",
      "                      are automatically sorted by read names.\n",
      "\n",
      "## Optional arguments:\n",
      "# Annotation\n",
      "\n",
      "  -F <string>         Specify format of the provided annotation file. Acceptable\n",
      "                      formats include 'GTF' (or compatible GFF format) and\n",
      "                      'SAF'. 'GTF' by default.  For SAF format, please refer to\n",
      "                      Users Guide.\n",
      "\n",
      "  -t <string>         Specify feature type(s) in a GTF annotation. If multiple\n",
      "                      types are provided, they should be separated by ',' with\n",
      "                      no space in between. 'exon' by default. Rows in the\n",
      "                      annotation with a matched feature will be extracted and\n",
      "                      used for read mapping. \n",
      "\n",
      "  -g <string>         Specify attribute type in GTF annotation. 'gene_id' by \n",
      "                      default. Meta-features used for read counting will be \n",
      "                      extracted from annotation using the provided value.\n",
      "\n",
      "  --extraAttributes   Extract extra attribute types from the provided GTF\n",
      "                      annotation and include them in the counting output. These\n",
      "                      attribute types will not be used to group features. If\n",
      "                      more than one attribute type is provided they should be\n",
      "                      separated by comma.\n",
      "\n",
      "  -A <string>         Provide a chromosome name alias file to match chr names in\n",
      "                      annotation with those in the reads. This should be a two-\n",
      "                      column comma-delimited text file. Its first column should\n",
      "                      include chr names in the annotation and its second column\n",
      "                      should include chr names in the reads. Chr names are case\n",
      "                      sensitive. No column header should be included in the\n",
      "                      file.\n",
      "\n",
      "# Level of summarization\n",
      "\n",
      "  -f                  Perform read counting at feature level (eg. counting \n",
      "                      reads for exons rather than genes).\n",
      "\n",
      "# Overlap between reads and features\n",
      "\n",
      "  -O                  Assign reads to all their overlapping meta-features (or \n",
      "                      features if -f is specified).\n",
      "\n",
      "  --minOverlap <int>  Minimum number of overlapping bases in a read that is\n",
      "                      required for read assignment. 1 by default. Number of\n",
      "                      overlapping bases is counted from both reads if paired\n",
      "                      end. If a negative value is provided, then a gap of up\n",
      "                      to specified size will be allowed between read and the\n",
      "                      feature that the read is assigned to.\n",
      "\n",
      "  --fracOverlap <float> Minimum fraction of overlapping bases in a read that is\n",
      "                      required for read assignment. Value should be within range\n",
      "                      [0,1]. 0 by default. Number of overlapping bases is\n",
      "                      counted from both reads if paired end. Both this option\n",
      "                      and '--minOverlap' option need to be satisfied for read\n",
      "                      assignment.\n",
      "\n",
      "  --fracOverlapFeature <float> Minimum fraction of overlapping bases in a\n",
      "                      feature that is required for read assignment. Value\n",
      "                      should be within range [0,1]. 0 by default.\n",
      "\n",
      "  --largestOverlap    Assign reads to a meta-feature/feature that has the \n",
      "                      largest number of overlapping bases.\n",
      "\n",
      "  --nonOverlap <int>  Maximum number of non-overlapping bases in a read (or a\n",
      "                      read pair) that is allowed when being assigned to a\n",
      "                      feature. No limit is set by default.\n",
      "\n",
      "  --nonOverlapFeature <int> Maximum number of non-overlapping bases in a feature\n",
      "                      that is allowed in read assignment. No limit is set by\n",
      "                      default.\n",
      "\n",
      "  --readExtension5 <int> Reads are extended upstream by <int> bases from their\n",
      "                      5' end.\n",
      "\n",
      "  --readExtension3 <int> Reads are extended upstream by <int> bases from their\n",
      "                      3' end.\n",
      "\n",
      "  --read2pos <5:3>    Reduce reads to their 5' most base or 3' most base. Read\n",
      "                      counting is then performed based on the single base the \n",
      "                      read is reduced to.\n",
      "\n",
      "# Multi-mapping reads\n",
      "\n",
      "  -M                  Multi-mapping reads will also be counted. For a multi-\n",
      "                      mapping read, all its reported alignments will be \n",
      "                      counted. The 'NH' tag in BAM/SAM input is used to detect \n",
      "                      multi-mapping reads.\n",
      "\n",
      "# Fractional counting\n",
      "\n",
      "  --fraction          Assign fractional counts to features. This option must\n",
      "                      be used together with '-M' or '-O' or both. When '-M' is\n",
      "                      specified, each reported alignment from a multi-mapping\n",
      "                      read (identified via 'NH' tag) will carry a fractional\n",
      "                      count of 1/x, instead of 1 (one), where x is the total\n",
      "                      number of alignments reported for the same read. When '-O'\n",
      "                      is specified, each overlapping feature will receive a\n",
      "                      fractional count of 1/y, where y is the total number of\n",
      "                      features overlapping with the read. When both '-M' and\n",
      "                      '-O' are specified, each alignment will carry a fractional\n",
      "                      count of 1/(x*y).\n",
      "\n",
      "# Read filtering\n",
      "\n",
      "  -Q <int>            The minimum mapping quality score a read must satisfy in\n",
      "                      order to be counted. For paired-end reads, at least one\n",
      "                      end should satisfy this criteria. 0 by default.\n",
      "\n",
      "  --splitOnly         Count split alignments only (ie. alignments with CIGAR\n",
      "                      string containing 'N'). An example of split alignments is\n",
      "                      exon-spanning reads in RNA-seq data.\n",
      "\n",
      "  --nonSplitOnly      If specified, only non-split alignments (CIGAR strings do\n",
      "                      not contain letter 'N') will be counted. All the other\n",
      "                      alignments will be ignored.\n",
      "\n",
      "  --primary           Count primary alignments only. Primary alignments are \n",
      "                      identified using bit 0x100 in SAM/BAM FLAG field.\n",
      "\n",
      "  --ignoreDup         Ignore duplicate reads in read counting. Duplicate reads \n",
      "                      are identified using bit Ox400 in BAM/SAM FLAG field. The \n",
      "                      whole read pair is ignored if one of the reads is a \n",
      "                      duplicate read for paired end data.\n",
      "\n",
      "# Strandness\n",
      "\n",
      "  -s <int or string>  Perform strand-specific read counting. A single integer\n",
      "                      value (applied to all input files) or a string of comma-\n",
      "                      separated values (applied to each corresponding input\n",
      "                      file) should be provided. Possible values include:\n",
      "                      0 (unstranded), 1 (stranded) and 2 (reversely stranded).\n",
      "                      Default value is 0 (ie. unstranded read counting carried\n",
      "                      out for all input files).\n",
      "\n",
      "# Exon-exon junctions\n",
      "\n",
      "  -J                  Count number of reads supporting each exon-exon junction.\n",
      "                      Junctions were identified from all the exon-spanning reads\n",
      "                      in the input (containing 'N' in CIGAR string). Counting\n",
      "                      results are saved to a file named '<output_file>.jcounts'\n",
      "\n",
      "  -G <string>         Provide the name of a FASTA-format file that contains the\n",
      "                      reference sequences used in read mapping that produced the\n",
      "                      provided SAM/BAM files. This optional argument can be used\n",
      "                      with '-J' option to improve read counting for junctions.\n",
      "\n",
      "# Parameters specific to paired end reads\n",
      "\n",
      "  -p                  If specified, libraries are assumed to contain paired-end\n",
      "                      reads. For any library that contains paired-end reads, the\n",
      "                      'countReadPairs' parameter controls if read pairs or reads\n",
      "                      should be counted.\n",
      "\n",
      "  --countReadPairs    If specified, fragments (or templates) will be counted\n",
      "                      instead of reads. This option is only applicable for\n",
      "                      paired-end reads. For single-end data, it is ignored.\n",
      "\n",
      "  -B                  Only count read pairs that have both ends aligned.\n",
      "\n",
      "  -P                  Check validity of paired-end distance when counting read \n",
      "                      pairs. Use -d and -D to set thresholds.\n",
      "\n",
      "  -d <int>            Minimum fragment/template length, 50 by default.\n",
      "\n",
      "  -D <int>            Maximum fragment/template length, 600 by default.\n",
      "\n",
      "  -C                  Do not count read pairs that have their two ends mapping \n",
      "                      to different chromosomes or mapping to same chromosome \n",
      "                      but on different strands.\n",
      "\n",
      "  --donotsort         Do not sort reads in BAM/SAM input. Note that reads from \n",
      "                      the same pair are required to be located next to each \n",
      "                      other in the input.\n",
      "\n",
      "# Number of CPU threads\n",
      "\n",
      "  -T <int>            Number of the threads. 1 by default.\n",
      "\n",
      "# Read groups\n",
      "\n",
      "  --byReadGroup       Assign reads by read group. \"RG\" tag is required to be\n",
      "                      present in the input BAM/SAM files.\n",
      "                      \n",
      "\n",
      "# Long reads\n",
      "\n",
      "  -L                  Count long reads such as Nanopore and PacBio reads. Long\n",
      "                      read counting can only run in one thread and only reads\n",
      "                      (not read-pairs) can be counted. There is no limitation on\n",
      "                      the number of 'M' operations allowed in a CIGAR string in\n",
      "                      long read counting.\n",
      "\n",
      "# Assignment results for each read\n",
      "\n",
      "  -R <format>         Output detailed assignment results for each read or read-\n",
      "                      pair. Results are saved to a file that is in one of the\n",
      "                      following formats: CORE, SAM and BAM. See Users Guide for\n",
      "                      more info about these formats.\n",
      "\n",
      "  --Rpath <string>    Specify a directory to save the detailed assignment\n",
      "                      results. If unspecified, the directory where counting\n",
      "                      results are saved is used.\n",
      "\n",
      "# Miscellaneous\n",
      "\n",
      "  --tmpDir <string>   Directory under which intermediate files are saved (later\n",
      "                      removed). By default, intermediate files will be saved to\n",
      "                      the directory specified in '-o' argument.\n",
      "\n",
      "  --maxMOp <int>      Maximum number of 'M' operations allowed in a CIGAR\n",
      "                      string. 10 by default. Both 'X' and '=' are treated as 'M'\n",
      "                      and adjacent 'M' operations are merged in the CIGAR\n",
      "                      string.\n",
      "\n",
      "  --verbose           Output verbose information for debugging, such as un-\n",
      "                      matched chromosome/contig names.\n",
      "\n",
      "  -v                  Output version of the program.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! featureCounts -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally we generate reports for all the steps with multiqc:\n",
    "\n",
    "multiqc /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/Alignment /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/featurecounts /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/fastqc /media/victor/c1d5c312-b546-4d5e-b24f-72dbe9e6f18f/SRR_files/Velasco_et_al/clean_data_reports/trimmed_files\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "rnaseq",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
